### PR TITLE
Prepare for splitting virt-v2v-wrapper into modules

### PR DESCRIPTION
**Pull request by nertpinx**
For now it's just an integration of setuptools for the basic things, lot of tests and other cahnges will need to be done, I'm sure, however I at least changed parts that I had time for.

The resulting binary is still called virt-v2v-wrapper.py, for back-compat purposes, but hopefully that will be fixed later.

This is still work in progress, but I'd appreciate any feedback.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: setup.py
Comment (nyoxi) [inline setup.py:27]: Travis fails on this with:
```
   error in virt-v2v-wrapper setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
```

**Review by nyoxi — CHANGES_REQUESTED** [review]
Generally it looks good. Let's get the tests fixed.

**Comment by nyoxi** [comment]
ci add to whitelist

File: setup.py
Comment (nertpinx) [reply setup.py:27]: OH, that's right, I just needed to move it out from the default requirements so that it is not installed when building the kubevirt container so that it does not drag in unnecessary things.  Then did not really try it out =)  Will fix that, thanks.

**Comment by nertpinx** [comment]
OK, phew, I made it work in a way that it passes all the checks.

* Good thing is that I think all the various build modes are handled now (unless I haven't found one or forgot to re-check some)
* Bad thing is that the resulting rpm package does not depend (although it should based on what I wrote int he spec file) on all the dependencies (pycurl, six) and I don't know where to look for the issue

**Comment by nertpinx** [comment]
One way to make sure the dependencies are there, for the time being, but not in the cleanest way possible, would be to just put them back into the spec file of course.

@nyoxi: What do you think?  Also are there more tests for this so that I can make sure everything works correctly? KubeVirt integration etc.?

File: v2v-conversion-host.spec.in
Comment (nyoxi) [inline v2v-conversion-host.spec.in]: This should not be dropped just yet.

File: setup.py
Comment (nyoxi) [inline setup.py]: Now that I checked the RPM content I see that this is wrong. Please drop the `.py` suffix. There is no need to leak the language in `/usr/bin`

File: v2v-conversion-host.spec.in
Comment (nyoxi) [inline v2v-conversion-host.spec.in]: Is this condition correct? Also, is the macro on the following line available in CentOS? If not we need to keep the deps static.

**Review by nyoxi — CHANGES_REQUESTED** [review]
It seems to me that your problem with dependencies is the condition around  `%{?python_enable_dependency_generator}`

File: v2v-conversion-host.spec.in
Comment (nertpinx) [reply v2v-conversion-host.spec.in]: It should be available everywhere, but since Fedora 30 it should be enabled by default. I wanted to mark it as such so that it can be removed later. I even tried removing the condition and it didn't help, so I think I'll just keep the requirements hard-coded in the rpm spec.

File: setup.py
Comment (nertpinx) [reply setup.py]: As noted in the commit message, I used this name to keep backwards compatibility for a while with consumers. If I can remove the extension I am even happier :smile:

File: v2v-conversion-host.spec.in
Comment (nertpinx) [reply v2v-conversion-host.spec.in]: OK, I'll keep it.

File: v2v-conversion-host.spec.in
Comment (nertpinx) [reply v2v-conversion-host.spec.in]: I hard-coded them for now, it can always be made nicer later.

**Review by nyoxi — COMMENTED** [review]
Do yo plan to do any changes? I think we can merge this.

**Comment by nertpinx** [comment]
> Do yo plan to do any changes? I think we can merge this.

Nope, I'm waiting for your review ;) It's good to go from my side.

**Review by nyoxi — APPROVED** [review]

**Pull request merged**
